pyop2/compilation.py

```diff
@@ -355,99 +383,127 @@ def get_so(self, jitmodule, extension):
             return ctypes.CDLL(soname)
 
 
-class MacCompiler(Compiler):
-    """A compiler for building a shared library on mac systems.
-
-    :arg cppargs: A list of arguments to pass to the C compiler
-         (optional).
-    :arg ldargs: A list of arguments to pass to the linker (optional).
-
-    :arg cpp: Are we actually using the C++ compiler?
+class MacClangCompiler(Compiler):
+    """A compiler for building a shared library on mac systems."""
+    _name = "Mac Clang"
 
-    :kwarg comm: Optional communicator to compile the code on (only
-        rank 0 compiles code) (defaults to COMM_WORLD).
-    """
+    _cflags = ["-fPIC", "-Wall", "-framework", "Accelerate", "-std=gnu11"]
+    _cxxflags = ["-fPIC", "-Wall", "-framework", "Accelerate"]
+    _ldflags = ["-dynamiclib"]
 
-    def __init__(self, cppargs=[], ldargs=[], cpp=False, comm=None):
+    @property
+    def _optflags(self):
         machine = platform.uname().machine
         opt_flags = ["-O3", "-ffast-math"]
         if machine == "arm64":
```

Comment:
I feel like we should have separate Arm and x86 compiler classes so we don't need this here.